docs/public-api.md

```diff
@@ -0,0 +1,97 @@
+# SwiftNIO 2 Public API
+
+The SwiftNIO project (which includes all of `github.com/apple/swift-nio*`) aims to follow [SemVer 2.0.0](https://semver.org/spec/v2.0.0.html) which requires projects to declare a public API. This document along with the [API documentation](https://apple.github.io/swift-nio/) declare SwiftNIO's API.
+
+## What are acceptable uses of NIO's Public API
+
+### Type, methods and modules
+
+It is acceptable and expected to use any type and call any method from the `NIO*` modules unless the type or function name start with an underscore (`_`). For `init`s this includes the first parameter's name.
```

Comment:
Should this specify explicitly "public" or "open" methods and types? For instance, using `@testable` import you could possibly get access to unstable methods not prefixed with `_`.